src/debugAdapter/debugNetwork.ts

```diff
@@ -97,4 +91,59 @@ export class DebugNetwork {
 
     return path.join(this.workingDirectory, directory);
   }
+
+  /**
+   * Gets all `LocalProject` from the `ItemType.LOCAL_SERVICE`.
+   *
+   * @returns an Array of `LocalProject` from the _Networks_ view.
+   */
+  private getProjects(): LocalProject[] {
+    const services = TreeManager.getItem(ItemType.LOCAL_SERVICE);
+
+    if (!services || !services.getChildren()) {
+      const error = new Error(Constants.ganacheCommandStrings.serverNoGanacheAvailable);
+      Telemetry.sendException(error);
+      throw error;
+    }
+
+    return services.getChildren() as LocalProject[];
+  }
+
+  /**
+   * Gets the `LocalNetworkNode` from a local project.
+   * If the `providerUrl` is present, it returns the `LocalNetworkNode` from that `provideUrl`
+   * Otherwise, it displays a `window.showQuickPick` to allow the user
+   * to select a `LocalNetworkNode` to start debugging.
+   *
+   * @param projects the tree items from the _Networks_ view to get `LocalNetworkNode`.
+   * @param providerUrl the network host url where the transaction has deployed, if any.
+   * @returns a promise that resolves to a `LocalNetworkNode`.
+   */
+  private async getHost(projects: LocalProject[], providerUrl?: string): Promise<LocalNetworkNode> {
+    if (providerUrl) {
+      return projects
+        .find((project) => {
+          const network = project.getChildren().at(0) as LocalNetworkNode;
+          const url = `${network.url.protocol}//${network.url.host}`;
+          return url === providerUrl;
+        })!
+        .getChildren()
+        .at(0) as LocalNetworkNode;
+    } else {
+      const items = projects.map((project) => {
+        return {
+          label: `$(callstack-view-session) ${project.label}`,
```

Comment:
Do you like the `$(debug-alt)` icon[1]? It's the same icon as _Run and Debug_ from VS Code.

---

[1] https://code.visualstudio.com/api/references/icons-in-labels

Reply:
Ok, we can change that, no worries